### ISSUE TITLE
after build reader_ops_test error

### ISSUE THREAD

**Issue by zhangjiulong** [tensorflow/models#107]
After I built the model I runned test like this
```
bazel test syntaxnet/... util/utf8/...
```
I got the error msg liek this:
```
AIL: //syntaxnet:reader_ops_test (see /home/zhangjl/.cache/bazel/_bazel_zhangjl/83e8305c713f72c9599677dec7a62070/syntaxnet/bazel-out/local-opt/testlogs/syntaxnet/reader_ops_test/test.log).
INFO: Elapsed time: 497.472s, Critical Path: 333.53s
//syntaxnet:arc_standard_transitions_test                                PASSED in 0.0s
//syntaxnet:beam_reader_ops_test                                         PASSED in 14.1s
//syntaxnet:graph_builder_test                                           PASSED in 9.4s
//syntaxnet:lexicon_builder_test                                         PASSED in 0.8s
//syntaxnet:parser_features_test                                         PASSED in 0.0s
//syntaxnet:parser_trainer_test                                          PASSED in 24.8s
//syntaxnet:sentence_features_test                                       PASSED in 0.1s
//syntaxnet:shared_store_test                                            PASSED in 0.2s
//syntaxnet:tagger_transitions_test                                      PASSED in 0.0s
//syntaxnet:text_formats_test                                            PASSED in 1.2s
//util/utf8:unicodetext_unittest                                         PASSED in 0.0s
//syntaxnet:reader_ops_test                                              FAILED in 3.2s
  /home/zhangjl/.cache/bazel/_bazel_zhangjl/83e8305c713f72c9599677dec7a62070/syntaxnet/bazel-out/local-opt/testlogs/syntaxnet/reader_ops_test/test.log
Executed 12 out of 12 tests: 11 tests pass and 1 fails locally.
There were tests whose specified size is too big. Use the --test_verbose_timeout_warnings command line option to see which ones these are.
```
then I added '--test_verbose_timeout_warnings' parameter like this
```
bazel test syntaxnet/... util/utf8/... --test_verbose_timeout_warnings
```
and i got the log like this
```
FAIL: //syntaxnet:reader_ops_test (see /home/zhangjl/.cache/bazel/_bazel_zhangjl/83e8305c713f72c9599677dec7a62070/syntaxnet/bazel-out/local-opt/testlogs/syntaxnet/reader_ops_test/test.log).
INFO: Elapsed time: 2.380s, Critical Path: 2.19s
//syntaxnet:arc_standard_transitions_test                       (cached) PASSED in 0.0s
//syntaxnet:beam_reader_ops_test                                (cached) PASSED in 14.1s
  WARNING: //syntaxnet:beam_reader_ops_test: Test execution time (0.0s excluding execution overhead) outside of range for MODERATE tests. Consider setting timeout="short" or size="small".
//syntaxnet:graph_builder_test                                  (cached) PASSED in 9.4s
  WARNING: //syntaxnet:graph_builder_test: Test execution time (0.0s excluding execution overhead) outside of range for MODERATE tests. Consider setting timeout="short" or size="small".
//syntaxnet:lexicon_builder_test                                (cached) PASSED in 0.8s
//syntaxnet:parser_features_test                                (cached) PASSED in 0.0s
//syntaxnet:parser_trainer_test                                 (cached) PASSED in 24.8s
  WARNING: //syntaxnet:parser_trainer_test: Test execution time (0.0s excluding execution overhead) outside of range for MODERATE tests. Consider setting timeout="short" or size="small".
//syntaxnet:sentence_features_test                              (cached) PASSED in 0.1s
  WARNING: //syntaxnet:sentence_features_test: Test execution time (0.0s excluding execution overhead) outside of range for MODERATE tests. Consider setting timeout="short" or size="small".
//syntaxnet:shared_store_test                                   (cached) PASSED in 0.2s
//syntaxnet:tagger_transitions_test                             (cached) PASSED in 0.0s
//syntaxnet:text_formats_test                                   (cached) PASSED in 1.2s
//util/utf8:unicodetext_unittest                                (cached) PASSED in 0.0s
  WARNING: //util/utf8:unicodetext_unittest: Test execution time (0.0s excluding execution overhead) outside of range for MODERATE tests. Consider setting timeout="short" or size="small".
//syntaxnet:reader_ops_test                                              FAILED in 2.2s
  /home/zhangjl/.cache/bazel/_bazel_zhangjl/83e8305c713f72c9599677dec7a62070/syntaxnet/bazel-out/local-opt/testlogs/syntaxnet/reader_ops_test/test.log

Executed 1 out of 12 tests: 11 tests pass and 1 fails locally.
```
i checked the log file'/home/zhangjl/.cache/bazel/_bazel_zhangjl/83e8305c713f72c9599677dec7a62070/syntaxnet/bazel-out/local-opt/testlogs/syntaxnet/reader_ops_test/test.log'
but i found nothing useful
What the matter?
[test.txt](https://github.com/tensorflow/models/files/267235/test.txt)

**Comment by galenmarchetti** [comment]
I have the same problem. 

` WARNING: //syntaxnet:reader_ops_test: Test execution time (0.0s excluding execution overhead) outside of range for MODERATE tests. Consider setting timeout="short" or size="small".
//syntaxnet:sentence_features_test                              (cached) PASSED in 1.4s
  WARNING: //syntaxnet:sentence_features_test: Test execution time (0.0s excluding execution overhead) outside of range for MODERATE tests. Consider setting timeout="short" or size="small".
//syntaxnet:shared_store_test                                   (cached) PASSED in 4.8s
//syntaxnet:tagger_transitions_test                             (cached) PASSED in 2.8s
//syntaxnet:text_formats_test                                   (cached) PASSED in 12.2s
//util/utf8:unicodetext_unittest                                (cached) PASSED in 3.3s
  WARNING: //util/utf8:unicodetext_unittest: Test execution time (0.0s excluding execution overhead) outside of range for MODERATE tests. Consider setting timeout="short" or size="small".
//syntaxnet:parser_trainer_test                                          FAILED in 0.1s
  /private/var/tmp/_bazel_gmarchetti/085ee26bdf8a2d54a89c4a8b04b3fedc/syntaxnet/bazel-out/local-opt/testlogs/syntaxnet/parser_trainer_test/test.log

Executed 1 out of 12 tests: 11 tests pass and 1 fails locally.`

In the log, I see that some files don't seem to be copied:

`+ TMP_DIR=/tmp/syntaxnet-output
+ mkdir -p /tmp/syntaxnet-output
+ sed s=SRCDIR=/private/var/tmp/_bazel_gmarchetti/085ee26bdf8a2d54a89c4a8b04b3fedc/syntaxnet/bazel-out/local-opt/bin/syntaxnet/parser_trainer_test.runfiles= /private/var/tmp/_bazel_gmarchetti/085ee26bdf8a2d54a89c4a8b04b3fedc/syntaxnet/bazel-out/local-opt/bin/syntaxnet/parser_trainer_test.runfiles/syntaxnet/testdata/context.pbtxt
+ sed s=OUTPATH=/tmp/syntaxnet-output=
sed: /private/var/tmp/_bazel_gmarchetti/085ee26bdf8a2d54a89c4a8b04b3fedc/syntaxnet/bazel-out/local-opt/bin/syntaxnet/parser_trainer_test.runfiles/syntaxnet/testdata/context.pbtxt: No such file or directory
+ PARAMS=128-0.08-3600-0.9-0
+ /private/var/tmp/_bazel_gmarchetti/085ee26bdf8a2d54a89c4a8b04b3fedc/syntaxnet/bazel-out/local-opt/bin/syntaxnet/parser_trainer_test.runfiles/syntaxnet/parser_trainer --arg_prefix=brain_parser --batch_size=32 --compute_lexicon --decay_steps=3600 --graph_builder=greedy --hidden_layer_sizes=128 --learning_rate=0.08 --momentum=0.9 --output_path=/tmp/syntaxnet-output --task_context=/tmp/syntaxnet-output/context --training_corpus=training-corpus --tuning_corpus=tuning-corpus --params=128-0.08-3600-0.9-0 --num_epochs=12 --report_every=100 --checkpoint_every=1000 --logtostderr
syntaxnet/parser_trainer_test: line 36: /private/var/tmp/_bazel_gmarchetti/085ee26bdf8a2d54a89c4a8b04b3fedc/syntaxnet/bazel-out/local-opt/bin/syntaxnet/parser_trainer_test.runfiles/syntaxnet/parser_trainer: No such file or directory`

**Comment by galenmarchetti** [comment]
I'm on OS X Yosemite 10.10.5

**Comment by Sondsara** [comment]
Getting almost the exact same error in the log as galenmarchetti: 

> exec ${PAGER:-/usr/bin/less} "$0" || exit 1
-----------------------------------------------------------------------------
+ BINDIR=/home/chris/.cache/bazel/_bazel_chris/49d59ff6167686f19b2bf01e6f762c6c/syntaxnet/bazel-out/local-opt/bin/syntaxnet/parser_trainer_test.runfiles/syntaxnet
+ CONTEXT=/home/chris/.cache/bazel/_bazel_chris/49d59ff6167686f19b2bf01e6f762c6c/syntaxnet/bazel-out/local-opt/bin/syntaxnet/parser_trainer_test.runfiles/syntaxnet/testdata/context.pbtxt
+ TMP_DIR=/tmp/syntaxnet-output
+ mkdir -p /tmp/syntaxnet-output
+ sed s=SRCDIR=/home/chris/.cache/bazel/_bazel_chris/49d59ff6167686f19b2bf01e6f762c6c/syntaxnet/bazel-out/local-opt/bin/syntaxnet/parser_trainer_test.runfiles= /home/chris/.cache/bazel/_bazel_chris/49d59ff6167686f19b2bf01e6f762c6c/syntaxnet/bazel-out/local-opt/bin/syntaxnet/parser_trainer_test.runfiles/syntaxnet/testdata/context.pbtxt
+ sed s=OUTPATH=/tmp/syntaxnet-output=
sed: can't read /home/chris/.cache/bazel/_bazel_chris/49d59ff6167686f19b2bf01e6f762c6c/syntaxnet/bazel-out/local-opt/bin/syntaxnet/parser_trainer_test.runfiles/syntaxnet/testdata/context.pbtxt: No such file or directory
+ PARAMS=128-0.08-3600-0.9-0
+ /home/chris/.cache/bazel/_bazel_chris/49d59ff6167686f19b2bf01e6f762c6c/syntaxnet/bazel-out/local-opt/bin/syntaxnet/parser_trainer_test.runfiles/syntaxnet/parser_trainer --arg_prefix=brain_parser --batch_size=32 --compute_lexicon --decay_steps=3600 --graph_builder=greedy --hidden_layer_sizes=128 --learning_rate=0.08 --momentum=0.9 --output_path=/tmp/syntaxnet-output --task_context=/tmp/syntaxnet-output/context --training_corpus=training-corpus --tuning_corpus=tuning-corpus --params=128-0.08-3600-0.9-0 --num_epochs=12 --report_every=100 --checkpoint_every=1000 --logtostderr
syntaxnet/parser_trainer_test: line 36: /home/chris/.cache/bazel/_bazel_chris/49d59ff6167686f19b2bf01e6f762c6c/syntaxnet/bazel-out/local-opt/bin/syntaxnet/parser_trainer_test.runfiles/syntaxnet/parser_trainer: No such file or directory

The last line is the same.

This is on Ubuntu 14.04

**Comment by cede-va** [comment]
I feel like I am getting relatively the same errors as above

bazel test syntaxnet/... util/utf8/... --test_verbose_timeout_warnings
WARNING: /private/var/tmp/_bazel_cede/f0a0ab679b008114f564c7277ee9560e/external/tf/WORKSPACE:1: Workspace name in /private/var/tmp/_bazel_cede/f0a0ab679b008114f564c7277ee9560e/external/tf/WORKSPACE (@__main__) does not match the name given in the repository's definition (@tf); this will cause a build error in future versions.
WARNING: /private/var/tmp/_bazel_cede/f0a0ab679b008114f564c7277ee9560e/external/re2/WORKSPACE:1: Workspace name in /private/var/tmp/_bazel_cede/f0a0ab679b008114f564c7277ee9560e/external/re2/WORKSPACE (@__main__) does not match the name given in the repository's definition (@re2); this will cause a build error in future versions.
WARNING: /private/var/tmp/_bazel_cede/f0a0ab679b008114f564c7277ee9560e/external/grpc/WORKSPACE:1: Workspace name in /private/var/tmp/_bazel_cede/f0a0ab679b008114f564c7277ee9560e/external/grpc/WORKSPACE (@__main__) does not match the name given in the repository's definition (@grpc); this will cause a build error in future versions.
INFO: Found 65 targets and 12 test targets...
ERROR: /Users/Vendasta/Projects/JamTime/ParsyFace/models/syntaxnet/syntaxnet/BUILD:470:1: Linking of rule '//syntaxnet:tagger_transitions_test' failed: cc_wrapper.sh failed: error executing command external/local_config_cc/cc_wrapper.sh -o bazel-out/local-opt/bin/syntaxnet/tagger_transitions_test '-Wl,-rpath,$ORIGIN/../_solib_darwin/' -Lbazel-out/local-opt/bin/_solib_darwin ... (remaining 63 argument(s) skipped): com.google.devtools.build.lib.shell.BadExitStatusException: Process exited with status 1.
clang: warning: argument unused during compilation: '-pthread'
error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/install_name_tool: changing install names or rpaths can't be redone for: bazel-out/local-opt/bin/syntaxnet/tagger_transitions_test (for architecture x86_64) because larger updated load commands do not fit (the program must be relinked, and you may need to use -headerpad or -headerpad_max_install_names)
INFO: Elapsed time: 0.558s, Critical Path: 0.26s
//syntaxnet:arc_standard_transitions_test                       (cached) PASSED in 0.1s
//syntaxnet:parser_features_test                                (cached) PASSED in 0.1s
//syntaxnet:sentence_features_test                              (cached) PASSED in 0.1s
  WARNING: //syntaxnet:sentence_features_test: Test execution time (0.0s excluding execution overhead) outside of range for MODERATE tests. Consider setting timeout="short" or size="small".
//syntaxnet:beam_reader_ops_test                                      NO STATUS
//syntaxnet:graph_builder_test                                        NO STATUS
//syntaxnet:lexicon_builder_test                                      NO STATUS
//syntaxnet:parser_trainer_test                                       NO STATUS
//syntaxnet:reader_ops_test                                           NO STATUS
//syntaxnet:shared_store_test                                         NO STATUS
//syntaxnet:text_formats_test                                         NO STATUS
//util/utf8:unicodetext_unittest                                      NO STATUS

Executed 0 out of 12 tests: 3 tests pass, 1 fails to build and 8 were skipped.

Any suggestions would be helpful!

**Comment by girving** [comment]
@djweiss: It seems like there are a lot of syntaxnet compilation issues.  Could you scrub through them and dedup or comment?

**Comment by apogre** [comment]
Having Same issue!

**Comment by skyprince999** [comment]
same here. Running it on Ubuntu 14.04 LTS with 4vCPUs & 15GB memory
Built it using the Dockerfile and was able to build it!

**Comment by tc64** [comment]
Same issue here.

**Comment by prakhar21** [comment]
@tc64 @apogre : Are you running bazel version 0.2.2 ?

**Comment by skyprince999** [comment]
Try what @prakhar21 has suggested 

remove the installed version of bazel using either 

`rm -fr ~/.cache/bazel
`
or
`rm -fr ~/.bazel`
or
`rm -fr ~/.bazelrc`

Then download the correct installer from the bazel [website](https://github.com/bazelbuild/bazel/releases)
bazel version == 0.2.2b

Install it using 
```
chmod +x bazel-version-installer-os.sh
./bazel-version-installer-os.sh --user
```

Re-build syntaxnet using bazel==0.2.2b

**Comment by apogre** [comment]
@prakhar21 Build label 0.3.0

**Comment by prakhar21** [comment]
@apogre purge the version you have. Download and install 0.2.2 from https://github.com/bazelbuild/bazel/releases. It must work then.

**Comment by tc64** [comment]
@prakhar21 I'm running 0.3.0, will revert to 0.2.2...
